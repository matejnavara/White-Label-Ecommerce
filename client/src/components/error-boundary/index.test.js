import { render, mount } from "enzyme";
import React from "react";

import ErrorBoundary from "./index";

it("component renders correctly", () => {
  expect(
    render(
      <ErrorBoundary>
        <p>Test</p>
      </ErrorBoundary>
    )
  ).toMatchSnapshot();
});

it("component catches error correctly", () => {
  const Node = () => null;
  const wrapper = mount(
    <ErrorBoundary>
      <Node />
    </ErrorBoundary>
  );
  wrapper.find(Node).simulateError(new Error("test"));
  expect(wrapper.state()).toEqual({ hasErrored: true });
});
