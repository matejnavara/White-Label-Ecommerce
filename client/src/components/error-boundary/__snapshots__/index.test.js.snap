// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component renders correctly 1`] = `
initialize {
  "0": Node {
    "attribs": Object {},
    "children": Array [
      Node {
        "data": "Test",
        "next": null,
        "parent": [Circular],
        "prev": null,
        "type": "text",
      },
    ],
    "name": "p",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": Node {
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
    },
    "prev": null,
    "type": "tag",
    "x-attribsNamespace": Object {},
    "x-attribsPrefix": Object {},
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "xml": false,
  },
}
`;
